ear-production-suite-plugins/plugins/binaural_monitoring/src/bear_data_files.cpp

```diff
@@ -37,114 +30,152 @@ juce::File getCustomDataFileDirectory() {
       getBearDataFileDirectory().getChildFile(BEAR_DATA_FILE));
   updateAvailableFiles();
 }
 
+namespace {
+void setFromMetadata(DataFileManager::DataFile& dataFile) {
+  try {
+    auto md = bear::DataFileMetadata::read_from_file(
+        dataFile.fullPath.getFullPathName().toStdString());
+    if (md.has_metadata()) {
+      dataFile.label = md.get_label();
+      dataFile.description = md.get_description();
+      dataFile.isBearRelease = md.is_released();
+    }
+  }
+  catch(std::exception const&) {
+    /* 
+    `read_from_file` could throw if;
+     - the file does not exist, or otherwise unreadable
+     - the file is not a valid tensorfile
+     - the metadata within the file is not of the expected structure
+    In all these cases, and any other, silently skip metadata extraction
+    
+    Note that a file with bad metadata may still be a usable filter set
+    so we don't want to strike it off yet, nor is it worth doing deeper 
+    checks here... the user can still select it as the processor will 
+    catch unusable files (unreadable/invalid/corrupt etc) anyway when 
+    it tries to start BEAR with it, and will report the actual error.
+    */
+  }
+}
+
+void updateFile(juce::File const& file, DataFileManager::FileMap& file_map,
+                bool released) {
+  auto [it, success] = file_map.insert(
+      {file, DataFileManager::DataFile{.filename = file.getFileName(),
+                                       .fullPath = file,
+                                       .label = "",
+                                       .description = "",
+                                       .isBearRelease = released}});
+  setFromMetadata(it->second);
+}
+}  // namespace
+
+bool DataFileManager::onlyContainsDefault() const {
+  return customDataFiles.empty() && releasedDataFiles.size() == 1u;
+}
+
+bool DataFileManager::defaultIsSelected() const {
+  return selectedDataFile_.has_value() &&
+         releasedDataFiles.contains(selectedDataFile_->fullPath);
+}
+
 void DataFileManager::updateAvailableFiles() {
-  // Start a new vec and copy existing shared_ptrs of unchanged files 
-  // This retains shared_ptrs unlike clear() and rebuild
-  std::vector<std::shared_ptr<DataFileManager::DataFile>> dfs;
-  // Lookup tfs where we expect to find custom files
-  auto files = getCustomDataFileDirectory().findChildFiles(
-      juce::File::TypesOfFileToFind::findFiles,
-                                    false, "*.tf");
+  releasedDataFiles.clear();
+  customDataFiles.clear();
+
   // add our expected released files
-  for(auto const& bearReleaseFile : bearReleaseFiles_) {
-    if (bearReleaseFile.existsAsFile()) {
-      // note that in win, the released bear dir is the same as the custom
-      // dir, so we might have already found this with findChildFiles
-      files.addIfNotAlreadyThere(bearReleaseFile);
+  for (auto const& file : bearReleaseFiles_) {
+    if (file.existsAsFile()) {
+      updateFile(file, releasedDataFiles, true);
```

Comment:
We should really check the `isBearRelease` flag here too as overwriting the default file with something custom would still be presented as a genuine release otherwise. It's important that it's not.